engine/src/main/java/org/camunda/bpm/engine/impl/bpmn/deployer/BpmnDeployer.java

```diff
@@ -255,29 +260,63 @@ protected void removeObsoleteTimers(ProcessDefinitionEntity processDefinition) {
     }
   }
 
-  protected void removeObsoleteEventSubscriptions(ProcessDefinitionEntity processDefinition, ProcessDefinitionEntity latestProcessDefinition) {
-    // remove all subscriptions for the previous version
-    if (latestProcessDefinition != null) {
-      EventSubscriptionManager eventSubscriptionManager = getEventSubscriptionManager();
+  protected void removeObsoleteEventSubscriptions(ProcessDefinitionEntity newLatestProcessDefinition, ProcessDefinitionEntity latestProcessDefinition) {
+    List<EventSubscriptionEntity> orphanSubscriptions = getOrphanSubscriptionEvents(newLatestProcessDefinition);
+    if(!orphanSubscriptions.isEmpty()) { // remove orphan subscriptions if any
+      for (EventSubscriptionEntity eventSubscriptionEntity : orphanSubscriptions) {
+        log.debug("Deleting orphan event subscription for process definition {}: {}", newLatestProcessDefinition.getId(), eventSubscriptionEntity.getEventName());
```

Comment:
❌ Let's skip this log for now. It doesn't fit tot he bpmn parse logger as well.
```suggestion
```

Reply:
Indeed, forgot to remove this thanks.